### ISSUE TITLE
`__row_group_by_columns_group__` should be an exported value

### ISSUE THREAD

**Issue by jmeinerz** [mui/mui-x#14470]
### Steps to reproduce

Link to live example: Not applicable

This is a development experience bug, not a user-facing one.

### Current behavior

When using `@mui/x-data-grid-premium`, MUI creates a new column on a data grid when the defines a custom grouping.

That new column is filterable, and the filter's field is defined as `__row_group_by_columns_group__`. This string is not exported, and that makes it for a brittle developer experience. Updating MUI might silently break things. 

### Expected behavior

I want to be able to import a value special-cased by MUI for the column created by a user's grouping, so that I can handle it accordingly in my consuming application.

I want the application to exit fast and be explicit in case a MUI version bump makes my assumptions incorrect.

### Your environment

Not applicable

**Search keywords**: __row_group_by_columns_group__ group by filters

**Comment by flaviendelangle** [comment]
It is exported in the `GRID_ROW_GROUPING_SINGLE_GROUPING_FIELD` variable.

@michelengelen if you want to double check :pray:

**Comment by jmeinerz** [comment]
Thank you, @flaviendelangle ! I can confirm the variable does solve the issue. I'm sorry I didn't find it before opening this. Cheers :)

**Comment by michelengelen** [comment]
@flaviendelangle you are right ... this specific part is not well documented. I'll add this as a recipe issue to add an example for this. Thanks for pointing that out! 👍🏼
